src/clark_west.jl

```diff
@@ -0,0 +1,76 @@
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export ClarkWestTest
+
+struct ClarkWestTest <: ZTest
+    n::Int       # number of observations
+    xbar::Real   # estimated mean
+    stderr::Real # population standard error
+    z::Real      # t-statistic
+    μ0::Real     # mean under h_0
+end
+
+"""
+    ClarkWestTest(e1::AbstractVector{<:Real}, e2::AbstractVector{<:Real}; lookahead::Integer=1)
+
+Perform the Clark-West test of equal performance of two nested prediction models, in terms of the
+out-of-sample mean squared prediction errors.
+
+`e1` is a vector of forecasts from the smaller (nested) model, `e2` is a vector of forecast
+errors from the larger model, and `lookahead` is the number of steps ahead of the forecast.
+Typically, the null hypothesis is that the two models perform equally well (a two-sided test),
+but sometimes we test whether the larger model performs better, which is indicated by a
+positive test statistic, for instance, above 1.645 for the 5% significance level (right tail test).
+
+Implements: [`pvalue`](@ref)
+# References
+ * Clark, T. E., West, K. D. 2006, Using out-of-sample mean squared prediction errors to test
+   the martingale difference hypothesis. Journal of Econometrics, 135(1): 155–186.
+ * Clark, T. E., West, K. D. 2007, Approximately normal tests for equal predictive accuracy
+   in nested models. Journal of Econometrics, 138(1): 291–311.
+
+"""
+function ClarkWestTest(e1::AbstractVector{<:Real}, e2::AbstractVector{<:Real};
+                       lookahead::Integer=1)
+    length(e1) == length(e2) || throw(DimensionMismatch("inputs must have the same length"))
+    n            = length(e1)
+    d            = 2*e1.*(e1 - e2)
+    cw_cov       = HypothesisTests.autocov(d, collect(0:lookahead-1))
+    cw_var       = (cw_cov[1] + 2*sum(cw_cov[2:end]))/n
+    xbar         = mean(d)
+    stderr       = sqrt(cw_var)
+    statistic_cw = xbar/stderr
+    return ClarkWestTest(n, xbar, stderr, statistic_cw, 0.0)
```

Comment:
`Int` is better as a neutral zero as it doesn't force promotion of e.g. `Float32`:
```suggestion
    return ClarkWestTest(n, xbar, stderr, statistic_cw, 0)
```
But is it really useful to have this field? Is it required by the `ZTest` interface (if it exists at all)?

Reply:
only used in `population_param_of_interest`. Probably not needed

Reply:
It is on the ZTest interface 
```
struct OneSampleZTest <: ZTest
    n::Int       # number of observations
    xbar::Real   # estimated mean
    stderr::Real # population standard error
    z::Real      # z-statistic
    μ0::Real     # mean under h_0
end
 ```

Reply:
This isn't an interface, it's the definition of a concrete struct inheriting from `ZTest`.